Makefile

```diff
@@ -64,6 +65,7 @@ MAGMA_DIR ?= ../magma
 # If CUDA_DIR is not set, check for nvcc, or resort to /usr/local/cuda
 CUDA_DIR  ?= $(or $(patsubst %/,%,$(dir $(patsubst %/,%,$(dir \
                $(shell which nvcc 2> /dev/null))))),/usr/local/cuda)
+HIP_DIR ?= /opt/rocm/hip
```

Comment:
I might be misremembering (writing from phone now), but I thought they install links at `/opt/rocm/bin/hipcc`, in which case we only need to deal with one path?

Reply:
Sorry, I'm not sure I understand what you mean.  One path for what?  Do you mean to not have both `HIP_DIR` and `ROCM_DIR` used in the Makefile?

Reply:
Yeah, `/opt/rocm/bin/hipcc` exists and is what their packager puts in PATH.

One other question, is it supported to build the HIP backend with `nvcc`?

Reply:
Fixed in 6fa873947f91776cbd578dc882bb27765424b693 

As for your question, do you mean building in Nvidia/`nvcc` mode of hipcc for an Nvidia GPU?  Or building directly with `nvcc`?  The first one should theoretically work, though I haven't tried it.  The second one, I don't know, maybe with the right includes from the HIP headers?

Reply:
My experience (compiling HIP examples using `nvcc`) has been that using `nvcc` directly spews lots of warnings. I think using `hipcc` to target CUDA is primarily a curiosity given that we have dedicated CUDA backends, but it could be interesting (and at least good to know) if the performance was comparable to native CUDA. I'll resolve this conversation, but it'd be interesting to test at some point.